pkg/operator/ceph/object/user/controller.go

```diff
@@ -151,7 +171,37 @@ func (c *ObjectStoreUserController) createUser(context *clusterd.Context, u *cep
 		DisplayName: &displayName,
 	}
 	objContext := object.NewContext(context, u.Spec.Store, u.Namespace)
-
+	// Wait with exponential backoff for the objectstore to be created
+	err := wait.ExponentialBackoff(backoff, func() (ok bool, err error) {
+		// check if CephObjectStore CR is created
+		logger.Infof("waiting for CephObjectStore %s to be created", u.Spec.Store)
+		_, err = context.RookClientset.CephV1().CephObjectStores(u.Namespace).Get(u.Spec.Store, metav1.GetOptions{})
+		// wait only if objectStore is not found. return on any other error
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			return true, fmt.Errorf("CephObjectStore %s could not be found. %+v", u.Spec.Store, err)
+		}
+		// check if ObjectStore is initialized
+		// rook does this by starting the RGW pod(s)
+		selector := fmt.Sprintf("%s=%s,%s=%s",
+			"rgw", u.Spec.Store,
+			"app", "rook-ceph-rgw",
+		)
+		logger.Infof("waiting for CephObjectStore %s to be initialized", u.Spec.Store)
+		_, err = context.Clientset.CoreV1().Pods(u.Namespace).List(metav1.ListOptions{LabelSelector: selector})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			return true, fmt.Errorf("CephObjectStore %s may not be initialized. %+v", u.Spec.Store, err)
+		}
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed while waiting for objectstore %s to be ready. %+v", u.Spec.Store, err)
```

Comment:
How about if we just log the error, and allow the method to proceed and attempt to create the user? This wait method isn't technically critical for the user to be created.

Reply:
I thought it was. Because user creation depends on creation of object store?
If we just wait for sometime and proceed anyway and the creation still fails, isn't that bad?

Reply:
User creation will likely fail if this fails, but at least the operator can attempt the creation before giving up.

Reply:
ack.